ocf_datapipes/load/pv/live.py

```diff
@@ -250,15 +250,16 @@ def get_pv_power_from_database(
     for provider in pv_output, solar_sheffield_passiv:
         idx = pv_yields_df["provider"] == provider
 
-        pv_yields_df.loc[idx, "pv_system_id"] = encode_label(
-            pv_yields_df.loc[idx, "pv_system_id"], label=provider
-        )
+        # If not idx.any() we try to assign to no indices and the data type is changed from int to
+        # float. Avoid this with if statement.
+        if idx.any():
+            pv_yields_df.loc[idx, "pv_system_id"] = encode_label(
+                pv_yields_df.loc[idx, "pv_system_id"], label=provider
+            )
```

Comment:
do we need a `else` with a loggin statement staying, tjere are not idx for prover ..., therefore we have not encoded the label or something like that

Reply:
Yeh sure, I'll add it